src/routers/alpha-router/alpha-router.ts

```diff
@@ -980,6 +974,39 @@ export class AlphaRouter
       estimatedGasUsedUSD,
     } = swapRouteRaw;
 
+    if (this.routeCachingProvider && cacheMode != CacheMode.Darkmode && swapRouteFromChain) {
+      // Generate the object to be cached
+      const routesToCache = CachedRoutes.fromRoutesWithValidQuotes(
+        swapRouteFromChain.routes,
+        this.chainId,
+        tokenIn,
+        tokenOut,
+        protocols.sort(), // sort it for consistency in the order of the protocols.
+        await blockNumber,
+        tradeType
+      );
+
+      if (routesToCache) {
+        // Attempt to insert the entry in cache. This is fire and forget promise.
+        // The catch method will prevent any exception from blocking the normal code execution.
+        this.routeCachingProvider.setCachedRoute(routesToCache, amount).then((success) => {
```

Comment:
I actually don't think I can do this, I'm avoiding having an `await` purposely here to do a "fire and forget", I don't want to block execution on insertion to cache.

Reply:
Ah I see. Well I think that means we won't guarantee that the cache is updated (see 'Background processes..' section [here](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtime-environment.html)). Maybe thats desirable? But one other option would be to just create the promise here, and `await` it later in the code when its more likely to have completed

Reply:
Yup, we won't guarantee that the cache is updated but that should be okay, inserting into dynamodb is a relatively quick operation, but I think in case the table is down or slow for any reason, we shouldn't block the request on it and we will let a different request do the write to cache.